samples/bluetooth/peripheral_cts_client/src/main.c

```diff
@@ -0,0 +1,372 @@
+#include <bluetooth/gatt_dm.h>
+#include <bluetooth/services/cts_client.h>
+
+#include <settings/settings.h>
+
+#include <dk_buttons_and_leds.h>
+
+#define DEVICE_NAME CONFIG_BT_DEVICE_NAME
+#define DEVICE_NAME_LEN (sizeof(DEVICE_NAME) - 1)
+
+#define RUN_STATUS_LED DK_LED1
+#define CON_STATUS_LED DK_LED2
+#define RUN_LED_BLINK_INTERVAL 1000
+
+#define KEY_READ_TIME DK_BTN1_MSK
+
+static struct bt_cts_client cts_c;
+
+static bool has_cts;
+static bt_security_t security_level;
+
+static const struct bt_data ad[] = {
+	BT_DATA_BYTES(BT_DATA_GAP_APPEARANCE,
+		      (CONFIG_BT_DEVICE_APPEARANCE >> 0) & 0xff,
+		      (CONFIG_BT_DEVICE_APPEARANCE >> 8) & 0xff),
+	BT_DATA_BYTES(BT_DATA_FLAGS, (BT_LE_AD_LIMITED | BT_LE_AD_NO_BREDR)),
+	BT_DATA_BYTES(BT_DATA_SOLICIT16, 0x05, 0x18),
+	BT_DATA(BT_DATA_NAME_COMPLETE, DEVICE_NAME, DEVICE_NAME_LEN),
+};
+
+static const char *day_of_week[] = { "Unknown",	  "Monday",   "Tuesday",
+				     "Wednesday", "Thursday", "Friday",
+				     "Saturday",  "Sunday" };
+
+static const char *month_of_year[] = { "Unknown",   "January", "February",
+				       "March",	    "April",   "May",
+				       "June",	    "July",    "August",
+				       "September", "October", "November",
+				       "December" };
+
+static void current_time_print(struct bt_cts_current_time *current_time)
+{
+	printk("\nCurrent Time:\n");
+	printk("\nDate:\n");
+
+	printk("\tDay of week   %s\n",
+	       day_of_week[current_time->exact_time_256.day_of_week]);
+
+	if (current_time->exact_time_256.day == 0) {
+		printk("\tDay of month  Unknown\n");
+	} else {
+		printk("\tDay of month  %u\n",
+		       current_time->exact_time_256.day);
+	}
+
+	printk("\tMonth of year %s\n",
+	       month_of_year[current_time->exact_time_256.month]);
+	if (current_time->exact_time_256.year == 0) {
+		printk("\tYear          Unknown\n");
+	} else {
+		printk("\tYear          %u\n",
+		       current_time->exact_time_256.year);
+	}
+	printk("\nTime:\n");
+	printk("\tHours     %u\n", current_time->exact_time_256.hours);
+	printk("\tMinutes   %u\n", current_time->exact_time_256.minutes);
+	printk("\tSeconds   %u\n", current_time->exact_time_256.seconds);
+	printk("\tFractions %u/256 of a second\n",
+	       current_time->exact_time_256.fractions256);
+
+	printk("\nAdjust reason:\n");
+	printk("\tDaylight savings %x\n",
+	       current_time->adjust_reason.change_of_daylight_savings_time);
+	printk("\tTime zone        %x\n",
+	       current_time->adjust_reason.change_of_time_zone);
+	printk("\tExternal update  %x\n",
+	       current_time->adjust_reason.external_reference_time_update);
+	printk("\tManual update    %x\n",
+	       current_time->adjust_reason.manual_time_update);
+}
+
+static void notify_current_time_cb(struct bt_cts_client *cts_c,
+				   struct bt_cts_current_time *current_time)
+{
+	current_time_print(current_time);
+}
+
+static void enable_notifications(void)
+{
+	int err;
+
+	if (has_cts && security_level >= BT_SECURITY_L2) {
```

Comment:
It seems like you don't need a static variable `security_level`. You could use `bt_conn_get_security` function to check current security level.

Reply:
Done.